test/library_note.lean

```diff
@@ -0,0 +1,66 @@
+import Batteries.Util.LibraryNote
+import Batteries.Test.Internal.DummyLibraryNote2
+
+/--
+info: Note not found
+-/
+#guard_msgs in
+#help note "no note"
+
+/--
+info: library_note "Other"
+/--
```

Comment:
I don't see the utility of wrapping each note in `/-- ... -/`. How about just separating with a blank line or `---`?

Reply:
i think the benefit is that this is less likely to cause confusion as to where one note stops and another begins, and this mimics the way you'd find the note in the actual code.

Reply:
I guess my issue is that `/-- ... -/` has a specific meaning in Lean, which is not relevant here. So, for example, copy pasting the output in a Lean file would lead to multiple errors. How about `/- ... -/`? (No errors then, though that still seems visually cluttered.)

Reply:
We aren't doing this for any of the other `#help` commands, I think the docs should just be indented relative to the `library_note` header.

Reply:
I reflected on this a bit and I think `#help` subcommands should give output that is better than plain code lookup in some useful way. In many cases organization is the key difference but formatting is relevant too. I don't know what would be the best formatting for library notes. How about asking on Zulip what users would like to see for library notes?

Reply:
i asked, and the only suggestion it got was that they would like if it would display the notes with markdown. AFAIK such functionality doesn't exist yet, and i don't think creating this functionality is in scope for this PR...

Reply:
This suggests using markdown style for bulleted lists.
```
* This is a note.

  Paragraphs within a note are separated by a blank line and indented.

* This is another note.
```